packages/mobile/src/app/saga.ts

```diff
@@ -115,8 +117,28 @@ export function* appRemoteFeatureFlagSaga() {
   }
 }
 
+function parseValue(value: string) {
+  if (['true', 'false'].indexOf(value) >= 0) {
+    return value === 'true'
+  }
+  const number = parseFloat(value)
+  if (!isNaN(number)) {
+    return number
+  }
+  return value
+}
+
+function parseQuery(query: string) {
+  const decodedParams = new URLSearchParamsReal(decodeURIComponent(query))
+  const params: { [key: string]: any } = {}
+  for (const [key, value] of decodedParams.entries()) {
+    params[key] = parseValue(value)
+  }
+  return params
+}
+
```

Comment:
This won't work for all types, we should add a comment about the limitations of this method.